libcontainer/cgroups/fs/cpuset.go

```diff
@@ -95,6 +95,10 @@ func (s *CpusetGroup) ensureParent(current, root string) error {
 	if filepath.Clean(parent) == root {
 		return nil
 	}
+	// Avoid infinite recursion.
+	if parent == current {
+		return nil
+	}
```

Comment:
The case `--cgroup-parent=../../../../../hello/this/is/a/crash` will cause Docker to crash due to an infinite recursion error. While this is solved with my path cleaning code, it implies that there are base cases not accounted for. `root` isn't `/`, it's just a regular path. So if `path` is a not a subdirectory of `root`, this will break. I can make it panic (or return an error) if we ever hit the case that `root` is a subdirectory of `path`.